### PR TITLE
Feature time to peak

### PR DESCRIPTION
**Pull request by vincentvdschaft**
# Changes
- I implemented a time-to-peak offset in the beamformer. Taking the time to peak into account can yield slightly improved image quality (see image).
- The scan object now has an element `t_peak` which is an array containing the time-to-peak of each transmit waveform. I considered adding t_peak as a dependent parameter on the waveform, but decided not to so people can choose to manually set it.
- There is a function `compute_time_to_peak` that finds the peaks of a sampled pulse and a function `compute_time_to_peak_stack` to compute the `t_peak` for a stack of waveforms.

t_peak_comparison

> [!WARNING]  
> This PR contains changes from #125 and should be merged after #125.

# Solves
- #50 

# How to test
- Beamform some file.
- Beamform some file after setting `scan.t_peak` to some value (2/center_frequency can be a reasonable guess)
- Run the `compute_time_to_peak_stack` function and see if the results make sense.

## Release Notes

* **New Features**
  * Added time-to-peak computation utilities for analyzing waveform characteristics
  * Introduced new scan properties for accessing peak time values and current transmit selection counts

* **Improvements**
  * Enhanced beamforming calculations with integrated peak time adjustments for improved accuracy
  * Optimized waveform data storage format for more efficient file handling
  * Fixed documentation typos and improved parameter descriptions

* **Backwards Compatibility**
  * Added automatic detection and conversion of legacy waveform data formats with deprecation warnings

### CODE REVIEW

File: zea/scan.py [hunk 1]
```diff
@@ -523,6 +538,15 @@ def initial_times(self):
 
         return value[self.selected_transmits]
 
+    @property
+    def t_peak(self):
+        """The time of the peak of the pulse in seconds of shape (n_tw,)."""
+        t_peak = self._params.get("t_peak")
+        if t_peak is None:
+            t_peak = np.array([1 / self.center_frequency])
```
Comment (OisinNolan) [inline zea/scan.py]: is `1/center_frequency` a kind of default value for the time to peak? If so, why is the default from `compute_time_to_peak` set to `0.0`?

**Review by OisinNolan — APPROVED** [review]
Nice addition! 🚀 definitely notice less blurring. Just left some minor comments

File: zea/scan.py (hunk 1 above)
Comment (vincentvdschaft) [reply zea/scan.py]: `1/center_frequency` is the period of the carrier, so I think it is a reasonable guess for the time to peak. The reasoning to have 0.0 as default argument in `compute_time_to_peak` that the peak of a pulse of `N` samples is always between 0.0 and `N/sampling_frequency`, therefore the peak of a signal of 0 samples should be zero. However, now that I think about it, it might be better to just raise an error as this is clearly a degenerate case. What do you think?

File: zea/data/file.py [hunk 2]
```diff
@@ -793,3 +794,71 @@ def _assert_unit_and_description_present(hdf5_file, _prefix=""):
...
+        #     5, 1, 23
+        # ]
+
+    Args:
+        dictionary (dict): The dictionary to convert.
+
+    Returns:
+        list: The sorted list of values.
+    """
+    return [value for _, value in sorted(dictionary.items())]
```
[the remark this answers is not in the document]
Comment (vincentvdschaft) [reply zea/data/file.py]: Good suggestion. I changed it.

File: zea/data/file.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 import h5py
 import numpy as np
+from keras.utils import pad_sequences
```
[the remark this answers is not in the document]
Comment (vincentvdschaft) [reply zea/data/file.py]: Keras is already a dependency.
